kitsune/search/v2/documents.py

```diff
@@ -242,3 +243,46 @@ def prepare_groups(self, instance):
     @classmethod
     def get_model(cls):
         return Profile
+
+
+class PostInnerDoc(InnerDoc):
+    content = field.Text()
+    author_id = field.Keyword()
+    created = field.Date()
+    updated = field.Date()
+    updated_by_id = field.Keyword()
+
+    @classmethod
+    def prepare(cls, instance):
+        return cls(
+            content=instance.content,
+            author_id=instance.author_id,
+            created=instance.created,
+            updated=instance.updated,
+            updated_by_id=instance.updated_by_id if instance.updated_by_id else None,
+        )
+
+
+class ForumThreadDocument(SumoDocument):
+    title = field.Text()
+    forum_id = field.Keyword()
+    created = field.Date()
+    creator_id = field.Keyword()
+    is_locked = field.Boolean()
+    is_sticky = field.Boolean()
+
+    # using Object rather than Nested here for performance at the expense of specificity
+    # e.g. we can't search for a thread in which a user jsmith posted "hello world",
+    # but we can search for a thread in which "hello world" was posted and jsmith posted
+    posts = field.Object(PostInnerDoc, multi=True)
```

Comment:
Couldn't we completely denormalize here and add the `PostInnerDoc` completely flat in here?